src/settings/AlertWordScreen.js

```diff
@@ -0,0 +1,123 @@
+/* @flow strict-local */
+
+import React, { PureComponent } from 'react';
+import { View } from 'react-native';
+
+import type { Auth, AlertWordsState } from '../types';
+import { connect } from '../react-redux';
+import { Screen, Input } from '../common';
+import ZulipButton from '../common/ZulipButton';
+import { getAuth } from '../selectors';
+import addAlertWords from '../api/alert_words/addAlertWords';
+import AlertWordList from './AlertWordList';
+import removeAlertWords from '../api/alert_words/removeAlertWords';
+import SearchEmptyState from '../common/SearchEmptyState';
+import ErrorMsg from '../common/ErrorMsg';
+import { showToast } from '../utils/info';
+import * as logging from '../utils/logging';
+
+const styles = {
+  containerStyle: {
+    marginHorizontal: 7,
+    marginTop: 20,
+  },
+  addButtonStyle: {
+    marginTop: 10,
+  },
+};
+
+type State = {|
+  text: string,
+  progress: boolean,
+  errorMessage: string,
+|};
+
+type Props = $ReadOnly<{|
+  auth: Auth,
+  alertWords: AlertWordsState,
+|}>;
+
+/**
+ * Alert words are strings that will be highlighted in messageList
+ */
+class AlertWordScreen extends PureComponent<Props, State> {
+  state = {
+    text: '',
+    progress: false,
+    errorMessage: '',
+  };
+
+  handleTextChange = (text: string) => {
+    this.setState({
+      text,
+    });
```

Comment:
Editing the text should probably clear the persistent error-message, if that's kept.